pwndbg/aglib/disasm/__init__.py

```diff
@@ -308,8 +302,16 @@ def can_run_first_emulate() -> bool:
     try:
         from mmap import mmap, MAP_ANON, MAP_PRIVATE, PROT_EXEC, PROT_READ, PROT_WRITE  # isort:skip
 
+        access_mode = PROT_WRITE | PROT_READ | PROT_EXEC
+        if sys.platform == "darwin":
+            # On macOS (Darwin), creating mmap with 'rwx' permissions is blocked
+            # by System Integrity Protection (SIP), which prevents execution of code
+            # from writable and readable memory regions. Therefore, we restrict
+            # the access mode to 'rw' (read and write only).
+            access_mode = PROT_WRITE | PROT_READ
```

Comment:
How does UnicornEngine works on macOS/darwin then? Does it not allocate an RWX memory page on macOS? How do they JIT the code then?

Reply:
I think Unicorn works fine. If you need executable page, first you create `rw`, then change it with mprotect to `rx`.
Only `wx` is not permitted.

<img width="761" alt="image" src="https://github.com/user-attachments/assets/8630f140-76f3-4068-b7f7-c75fe9328c17">